src/Reflector/Generic.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace BetterReflection\Reflector;
+
+use BetterReflection\Reflection\Symbol;
+use BetterReflection\SourceLocator\LocatedSource;
+use BetterReflection\SourceLocator\SourceLocator;
+use BetterReflection\Reflection\ReflectionClass;
+use BetterReflection\Reflection\Reflection;
+use PhpParser\Parser;
+use PhpParser\Lexer;
+use PhpParser\Node;
+
+class Generic
+{
+    /**
+     * @var SourceLocator
+     */
+    private $sourceLocator;
+
+    public function __construct(SourceLocator $sourceLocator)
+    {
+        $this->sourceLocator = $sourceLocator;
+    }
+
+    /**
+     * Uses the SourceLocator given in the constructor to locate the $symbolName
+     * specified and returns the \Reflector
+     *
+     * @param Symbol $symbol
+     * @return Reflection
+     */
+    public function reflect($symbol)
+    {
+        if (!($symbol instanceof Symbol)) {
```

Comment:
You are not checking for the symbol type. Did you consider splitting `ClassSymbol` and other symbols by type, rather than property?